src/components/CheckoutBilling/index.tsx

```diff
@@ -1,135 +1,163 @@
+import * as H from "history";
 import * as React from "react";
-import { Mutation } from "react-apollo";
 import { RouteComponentProps } from "react-router";
 import { Link } from "react-router-dom";
 
 import { AddressSummary, ShippingAddressForm } from "..";
-import { CheckoutContext } from "../CheckoutApp/context";
+import { maybe } from "../../core/utils";
+import {
+  CheckoutContext,
+  CheckoutContextInterface
+} from "../CheckoutApp/context";
 import {
   checkoutBaseUrl,
   checkoutPaymentUrl,
   checkoutShippingOptionsUrl
 } from "../CheckoutApp/routes";
+import {
+  Checkout,
+  Checkout_billingAddress
+} from "../CheckoutApp/types/Checkout";
 import { ShopContext } from "../ShopProvider/context";
-import { UPDATE_CHECKOUT_BILLING_ADDRESS } from "./queries";
+import { getShop_shop } from "../ShopProvider/types/getShop";
+import { TypedUpdateCheckoutBillingAddressMutation } from "./queries";
+import { updateCheckoutBillingAddress } from "./types/updateCheckoutBillingAddress";
 
-class CheckoutBilling extends React.Component<RouteComponentProps<{ id }>, {}> {
-  render() {
-    return (
-      <div>
-        <CheckoutContext.Consumer>
-          {({ checkout, updateCheckout, shippingAsBilling }) => {
-            const address =
-              !checkout.billingAddress && shippingAsBilling
-                ? checkout.shippingAddress
-                : checkout.billingAddress;
-            return (
-              <>
-                <Link to={checkoutBaseUrl(checkout.token)}>
-                  <div className="checkout__step checkout__step--inactive">
-                    <span>1</span>
-                    <h4 className="checkout__header">Shipping Address</h4>
-                  </div>
-                </Link>
-                <div className="checkout__content">
-                  <AddressSummary
-                    address={checkout.shippingAddress}
-                    email={checkout.email}
-                  />
-                </div>
-                <Link to={checkoutShippingOptionsUrl(checkout.token)}>
-                  <div className="checkout__step checkout__step--inactive">
-                    <span>2</span>
-                    <h4 className="checkout__header">Shipping Method</h4>
-                  </div>
-                </Link>
+const proceedToPayment = (
+  data: updateCheckoutBillingAddress,
+  checkoutCtx: CheckoutContextInterface,
+  history: H.History
+) => {
+  const canProceed = maybe(
+    () => !data.checkoutBillingAddressUpdate.errors.length
```

Comment:
AFAIK you don't need to put this statement in `maybe`, you should always have `.errors` property